ffi/src/lib.rs

```diff
@@ -1783,17 +1999,27 @@ pub unsafe extern "C" fn tx_change_list_entry_at(tx_report_list: *mut TxChangeLi
 /// If there is no value present at the `index`.
 #[no_mangle]
 pub unsafe extern "C" fn changelist_entry_at(tx_report: *mut TransactionChange, index: c_int) -> Entid {
+    assert_not_null!(tx_report);
     let tx_report = &*tx_report;
-    let index = index as usize;
-    tx_report.changes[index].clone()
+    assert!(0 <= index && (index as usize) < (tx_report.changes_len as usize));
+    std::ptr::read(tx_report.changes.offset(index as isize))
+}
+
+#[no_mangle]
+pub unsafe extern "C" fn destroy_mentat_string(s: *mut c_char) {
```

Comment:
Can this be `c_string_destroy` instead? That will keep the pattern of `<type>_destroy` we have for all of the other destructors and `<type>_action` that we have for all the FFI functions and de-mentatifies a function that applies to all C Strings and not just Mentat generated ones.

Reply:
> de-mentatifies a function that applies to all C Strings and not just Mentat generated ones.

I did this mostly because I wanted to avoid name collisions, but we can probably address them if/when they become an issue.